distribution/lib/Standard/Snowflake/0.0.0-dev/src/Internal/Snowflake_Type_Mapping.enso

```diff
@@ -0,0 +1,183 @@
+from Standard.Base import all
+import Standard.Base.Errors.Illegal_Argument.Illegal_Argument
+
+import Standard.Table.Data.Type.Value_Type.Bits
+import Standard.Table.Data.Type.Value_Type.Value_Type
+import Standard.Table.Internal.Java_Exports
+from Standard.Table.Errors import Inexact_Type_Coercion
+
+import Standard.Database.Internal.Column_Fetcher as Column_Fetcher_Module
+import Standard.Database.Internal.Column_Fetcher.Column_Fetcher
+import Standard.Database.Internal.IR.SQL_Expression.SQL_Expression
+import Standard.Database.Internal.SQL_Type_Mapping
+import Standard.Database.Internal.SQL_Type_Reference.SQL_Type_Reference
+import Standard.Database.SQL_Type.SQL_Type
+from Standard.Database.Errors import Unsupported_Database_Operation
+
+polyglot java import java.sql.Types
+
+## PRIVATE
+type Snowflake_Type_Mapping
+    ## PRIVATE
+    value_type_to_sql : Value_Type -> Problem_Behavior -> SQL_Type
+    value_type_to_sql value_type on_problems =
+        result = case value_type of
+            Value_Type.Boolean -> SQL_Type.Value Types.BOOLEAN "boolean"
+            # All integer types in Snowflake become NUMERIC(38,0).
+            Value_Type.Byte -> SQL_Type.Value Types.TINYINT "tinyint"
+            Value_Type.Integer Bits.Bits_16 -> SQL_Type.Value Types.SMALLINT "smallint"
+            Value_Type.Integer Bits.Bits_32 -> SQL_Type.Value Types.INTEGER "int"
+            Value_Type.Integer Bits.Bits_64 -> SQL_Type.Value Types.BIGINT "bigint"
+            # All float types in Snowflake become double
+            Value_Type.Float Bits.Bits_32 -> SQL_Type.Value Types.REAL "float4"
+            Value_Type.Float Bits.Bits_64 -> SQL_Type.Value Types.DOUBLE "float8"
```

Comment:
I'm not sure if it is worth lying here then?

Let's just convert any `Float` type to the true thing, i.e. `float8`.

```suggestion
            # All float types in Snowflake become double
            Value_Type.Float _ -> SQL_Type.Value Types.REAL "float8"
```

I think that is the whole point of this mapping logic that it tries to reflect what type is there really in the DB. It will add a warning when the type is coerced.

OR does my suggested change induce some spurious warnings I'm not aware of?